lib/puppet/pops/lookup.rb

```diff
@@ -56,6 +58,22 @@ def self.lookup(name, value_type, default_value, has_default, merge, lookup_invo
         lookup_invocation.emit_debug_info(debug_preamble(names)) if Puppet[:debug]
         fail_lookup(names)
       end
+    elsif !convert_to.nil?
+      convert_to = [convert_to] unless convert_to.is_a?(Array)
+      # TODO: if this fails the error needs to reference the lookup_option file/line
+      answer = lookup_invocation.scope.call_function(NEW, [convert_to[0], answer, *(convert_to[1..-1])])
+    else
+      opt_key = Puppet::Pops::Lookup::LookupKey.new(name)
+      options = lookup_invocation.lookup_adapter.lookup_lookup_options(opt_key, lookup_invocation)
+      ct = options[CONVERT_TO] unless options.nil?
+      if !ct.nil?
+        ct = ct.is_a?(Array) ? ct : [ct]
+        unless ct[0].is_a?(Puppet::Pops::Types::PTypeType)
+          ct[0] = Puppet::Pops::Types::TypeParser.singleton.parse(ct[0])
+        end
+        # TODO: if this fails the error needs to reference the lookup_option file/line
```

Comment:
I think that will be complicated because:
- The result of parsing a YAML is a hash. It doesn't contain file/line info.
- The lookup options is a result of merging several hashes.
- A solution will need to either re-read all configurations to figure out what file/line it was that provided this option, or we need a new parser where we conserve the AST and somehow annotate the found entries before they are merged and then retain that info throughout the merges.

Reply:
Maybe just say something that it "got it from lookup options"